### ISSUE TITLE
[java] CloseResource false positive with ByteArrayInputStream

### ISSUE THREAD

**Issue by oowekyala** [pmd/pmd#1921]
**Affects PMD Version:** 6.16
**Rule:** [CloseResource](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#closeresource)
**Description:**

The rule makes an exception for `ByteArrayOutputStream` but not the input stream. From the doc:

> Closing a ByteArrayInputStream has no effect. The methods in this class can be called after the stream has been closed without generating an IOException.

**Code Sample demonstrating the issue:**
```java
import java.io.ByteArrayInputStream;
import java.nio.charset.StandardCharsets;

class Foo {

    {
        ByteArrayInputStream bis = new ByteArrayInputStream("fooString".getBytes(StandardCharsets.UTF_8));
    }
}
```
